interfaces/builtin/login_session_control_test.go

```diff
@@ -0,0 +1,93 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type loginSessionControlSuite struct {
+	iface    interfaces.Interface
+	slotInfo *snap.SlotInfo
+	slot     *interfaces.ConnectedSlot
+	plugInfo *snap.PlugInfo
+	plug     *interfaces.ConnectedPlug
+}
+
+var _ = Suite(&loginSessionControlSuite{
+	iface: builtin.MustInterface("login-session-control"),
+})
+
+func (s *loginSessionControlSuite) SetUpTest(c *C) {
+	consumingSnapInfo := snaptest.MockInfo(c, `
+name: other
+version: 0
+apps:
+ app:
+    command: foo
+    plugs: [login-session-control]
+`, nil)
+	s.plugInfo = consumingSnapInfo.Plugs["login-session-control"]
+	s.plug = interfaces.NewConnectedPlug(s.plugInfo, nil, nil)
+	s.slotInfo = &snap.SlotInfo{
+		Snap:      &snap.Info{SuggestedName: "core", Type: snap.TypeOS},
+		Name:      "login-session-control",
+		Interface: "login-session-control",
+	}
+	s.slot = interfaces.NewConnectedSlot(s.slotInfo, nil, nil)
+}
+
+func (s *loginSessionControlSuite) TestName(c *C) {
+	c.Assert(s.iface.Name(), Equals, "login-session-control")
+}
+
+func (s *loginSessionControlSuite) TestSanitizeSlot(c *C) {
+	c.Assert(interfaces.BeforePrepareSlot(s.iface, s.slotInfo), IsNil)
+	slot := &snap.SlotInfo{
+		Snap:      &snap.Info{SuggestedName: "some-snap"},
+		Name:      "login-session-control",
+		Interface: "login-session-control",
+	}
+	c.Assert(interfaces.BeforePrepareSlot(s.iface, slot), ErrorMatches,
```

Comment:
(nitpick²) The usual convention in go is not to wrap lines, so this line and the following should be in a single line (but its fine to ignore for this PR, mostly a FYI).

Reply:
OK. FWIW I was just following existing practice:
```plain
interfaces/builtin/account_control_test.go:87
interfaces/builtin/alsa_test.go:75
interfaces/builtin/autopilot_test.go:77
...
```

Reply:
Heh, sorry for that then - looks like we are less consistent about it then I thought :)Thanks for clarifying this.